pkg/policy/groups/controllers.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groups
+
+import (
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+)
+
+const (
+	maxConcurrentUpdates = 4
+)
+
+func UpgradeCNPInformation() {
+	cnpToUpdate := groupsCNPCache.GetAllCNP()
+	sem := make(chan bool, maxConcurrentUpdates)
+	for _, cnp := range cnpToUpdate {
+		sem <- true
+		go func(cnp *cilium_v2.CiliumNetworkPolicy) {
+			defer func() { <-sem }()
```

Comment:
Isn't this the wrong way around? We put stuff into the sem channel, but this defer blocks the goroutine from exiting, but only after it did addDerivativeCNP and that's the thing we want to slow down.
In this case, it might be simpler to have the workers ready, and write all the CNPs in cnpToUpdate into a channel that they read from. Then, you can check a counter or something (like maybe read them all back from an output channel) to know they are ready.

Reply:
Umm, I'm not sure if I understand this one. The limit of calls to AWS is limited, so I made a function to limit to 4 in the same time,  you can slow down or you can wait until is free, is the same behaviour in a different way no?